File: betolyn/src/main/java/com/betolyn/features/auth/AuthService.java
package com.betolyn.features.auth;

import com.betolyn.features.auth.dtos.SignInRequestDTO;
import com.betolyn.features.auth.dtos.SignUpRequestDTO;
import lombok.RequiredArgsConstructor;

import java.util.Optional;

@RequiredArgsConstructor
public class AuthService implements IAuthService {
    private final UserRepository userRepository;

    @Override
    public void signUp(SignUpRequestDTO requestDTO) {
        var email = requestDTO.getEmail();
        var existsUser = Optional.ofNullable(userRepository.findByEmail(email));
        if (existsUser.isEmpty()) {
            return;
        }

        userRepository.save(new UserEntity("id", requestDTO.getPassword(), email));
    }

    @Override
    public void signIn(SignInRequestDTO requestDTO) {
        throw new UnsupportedOperationException();
    }
}

File: mobile/services/auth-service.ts
import { getRequest, postRequest } from '@/utils/http';
import {
  SignInRequestDTO,
  SignInResponseDTO,
  SignUpRequestDTO,
  SignUpResponseDTO,
} from '../app/auth/signup/types';

export class AuthService {
  public static async getMe() {
    return await getRequest<SignInResponseDTO>('/me');
  }

  public static async signIn(requestDTO: SignInRequestDTO) {
    return await postRequest<SignInResponseDTO, SignInRequestDTO>('/auth/signin', requestDTO);
  }

  public static async signUp(requestDTO: SignUpRequestDTO) {
    return await postRequest<SignUpResponseDTO, SignUpRequestDTO>('/auth/signup', requestDTO);
  }

  public static async signOut() {
    return await getRequest<void>(`/auth/logout`);
  }
}
